AlicerceTests/Extensions/OptionalTests.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  OptionalTests.swift
+//  Alicerce
+//
+//  Created by Luís Afonso on 07/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import XCTest
+
+@testable import Alicerce
+
+final class OptionalTests: XCTestCase {
+
+    func testThen_UsingNonOptionalValue_ShouldUnwrapTheValue() {
+        let anOptionalString: String? = "😎"
+
+        anOptionalString.then { value in
+            XCTAssertTrue(anOptionalString! == value, "🚀 \(value) unwrapped ✅")
```

Comment:
Sorry for the delay, @portellaa 😊 

What I suggested was slightly different from this. Even though this is a little bit better, we still don't know for sure that the block was effectively executed, right?

Some notes to fix that:
1. We can keep an assertion inside the `then` closure, like: 
`XCTAsserEqual(anOptionalString, value, "🔥: \(value) not unwrapped! 😱")`
(note that the string in this case is the message shown when the assert *fails* 😛 )

2. Change `value` *inside* the `then` closure, e.g.:  
`anOptionalString = "😇"`
3. Do an assert *after* the `then` closure to check that the value was properly modified, e.g.: 
`XCTAsserEqual(anOptionalString, "😇", "🔥: `then` closure not executed! 😭")`